pkg/vault/rule_test.go

```diff
@@ -0,0 +1,53 @@
+package vault_test
+
+import (
+	"testing"
+
+	"github.com/jenkins-x/jx/pkg/vault"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEncodeVaultPathRule(t *testing.T) {
+	tests := map[string]struct {
+		rule *vault.PathRule
+		err  bool
+		want string
+	}{
+		"marshal policy": {
+			rule: &vault.PathRule{
+				Path: vault.PathPolicy{
+					Prefix:       "secrets/*",
+					Capabilities: []string{vault.CreateCapability, vault.ReadCapability},
+				},
+			},
+			err:  false,
+			want: "path \"secrets/*\" {  capabilities = [    \"create\",    \"read\",  ]}",
+		},
+		"marshal empty policy": {
+			rule: &vault.PathRule{
+				Path: vault.PathPolicy{
+					Prefix:       "",
+					Capabilities: []string{},
+				},
+			},
+			err:  false,
+			want: "path \"\" {  capabilities = []}",
+		},
+		"marshal nil policy with error": {
+			rule: nil,
+			err:  true,
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			output, err := tc.rule.String()
```

Comment:
Don't compare strings for things like this. 
1) It causes a lot of unnecessary headaches (eg, if we change a library implementation at some point, or change whitespace formatting, the tests will fail). Also, testing like this broke on Windows where a newline is CR-LF but in unix it's just LF
2) Ultimately, we don't care about the `string` representation, we care about the `json`/`yaml` representation.

keep `tc.rule` as a struct (or a JSON/YAML object) and deserialize the the `want` to the same type, then do a comparison on the typed objects.